File: C/Sort search/Binary search.c
#include <stdio.h>
int main()
{
   int i,first,last,mid,size,search,flag = 0;
   int arr[100];

   printf("Enter number of elements...\n");
   scanf("%d",&size);
   
   printf("Enter the array in ascending order_ _ _\n",size);
   for(i=0;i<size;i++){
      scanf("%d",&arr[i]);
   }
   
   printf("Enter the number to be search...\n");
   scanf("%d",&search);
   
   first = 0;
   last = size - 1;
   mid = (first+last)/2;
   while(first <= last){
        if(arr[mid] < search)
           first = mid + 1;
           mid=(first + last)/2;
        if(arr[mid]==search){
        	flag = 1;
          printf("%d found at location %d.\n",search,mid+1);
          break;
       }
        if(arr[mid] > search)
         last = mid - 1;
        mid = (first + last)/2;
 }
     if(flag == 0)
         printf("Not found!! %d isn't present in the list.\n",search);
       return 0;
}

File: Python/binary search.py
def binarySearch(arrList,n):
	first = 0
	last = len(arrList)-1
	mid = (first+last)//2
	while first <= last:
		if arrList[mid] == n : return mid
		elif arrList[mid] > n:
			last = mid-1 
			mid = (first+last) // 2
		else:
			first = mid+1
			mid = (first+last) // 2
	return -1
	
def varify(num):
	if num <= -1 : print('Number is not found')
	else : print('Number is found at index {} '.format(num))

lis = [3,6,7,10,13,27,30,31,38,40,44,47,49,50]
varify(binarySearch(lis,31))